src/scheduler/queue_info.c

```diff
@@ -184,6 +188,32 @@ query_queues(status *policy, int pbs_sd, server_info *sinfo)
 			return NULL;
 		}
 
+		if (dflt_sched) {
+			if (qinfo->partition != NULL) {
+				cur_queue = cur_queue->next;
+				continue;
+			}
+		} else {
+			if (qinfo->partition == NULL) {
+				cur_queue = cur_queue->next;
+				continue;
+			} else {
+				int i;
+				int in_partition = 0;
+				for (i=0; my_partitions[i] != NULL; i++) {
+					if (strcmp(qinfo->partition, my_partitions[i]) == 0) {
+						in_partition = 1;
+						break;
+					}
+				}
+				if (!in_partition) {
+					cur_queue = cur_queue->next;
+					continue;
+				}
+
+			}
+		}
+
```

Comment:
Like my above comment, put this code in a function and avoid the use of continues

Reply:
Implemented.